plugin/core/transports.py

```diff
@@ -185,6 +185,12 @@ def close(self) -> None:
         self.send_queue.put(None)  # kill the write thread as it's blocked on send_queue
         self.on_closed()
 
+    def _checked_stdout(self) -> 'IO[Any]':
+        if self.process:
+            return self.process.stdout
+        else:
+            raise UnexpectedProcessExitError()
+
```

Comment:
I'm going to come across as super nitpicky now, but this is incorrect again. There's the possibility that an AttributeError will bubble up from this function.

The absolute sure way to handle this is

```python
def _checked_stdout(self) -> 'IO[Any]':
    try:
        return self.process.stdout
    except AttributeError as ex:
        raise UnexpectedProcessExitError() from ex
```

Reply:
Sure, but does it happen in your test now?